### PR TITLE
Get Calls Must not have the LRO properties

### PR DESCRIPTION
**Pull request by tejaswiMinnu**
This rule will check for "x-ms-long-running-operation-options" property block on the get calls.
I also added condition to check x-ms-long-running-operation set to true will flag an error.
https://github.com/Azure/azure-rest-api-specs-pr/pull/12758#discussion_r1205859448

### CODE REVIEW

File: common/changes/@microsoft.azure/openapi-validator-rulesets/tejaswis-getMustNotBeLRO_2023-09-19-23-56.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@microsoft.azure/openapi-validator-rulesets",
+      "comment": "Added GetMustNotBeLRO, this rule make sures get calls must not have LRO properties as get calls are synchronous",
```
Comment (AkhilaIlla) [inline common/changes/@microsoft.azure/openapi-validator-rulesets/tejaswis-getMustNotBeLRO_2023-09-19-23-56.json:5]: > make sures [](http://example.com/codeflow?start=50&length=11)

nit: makes sure

File: packages/rulesets/src/spectral/az-arm.ts [hunk 2]
```diff
@@ -351,6 +351,22 @@ const ruleset: any = {
       },
     },
 
+    // RPC Code: RPC-Get-V1-14
+    GetMustNotBeLRO: {
+      description: "The Get call cannot be Long Running Operation and it must not have x-ms-long-running-operation property.",
```
Comment (AkhilaIlla) [inline packages/rulesets/src/spectral/az-arm.ts:356]: >The Get call cannot be Long Running Operation and it must not have x-ms-long-running-operation property [](http://example.com/codeflow?start=20&length=103)

The GET operation call cannot be Long Running and MUST not have x-ms-long-running-operation property defined?

File: packages/rulesets/src/spectral/az-arm.ts [hunk 3]
```diff
@@ -351,6 +351,22 @@ const ruleset: any = {
       },
     },
 
+    // RPC Code: RPC-Get-V1-14
+    GetMustNotBeLRO: {
```
Comment (AkhilaIlla) [inline packages/rulesets/src/spectral/az-arm.ts:355]: >GetMustNotBeLRO [](http://example.com/codeflow?start=4&length=15)

GetOperationMustNotBeLongRunning ?

File: docs/get-must-not-be-lro.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+## Description
+
+Only asynchronous(i.e. Long Running Operation) can have `x-ms-long-running-operation-options` property.
+The Get calls are synchronous and it MUST NOT have
+    - `x-ms-long-running-operation-options` property block
+    - `x-ms-long-running-operation-options` set to `true`
+
+## How to fix the violation
+
+Ensure that the get operation `x-ms-long-running-operation` property set to `false` and MUST NOT contain `x-ms-long-running-operation-options` property block.
```
Comment (AkhilaIlla) [inline docs/get-must-not-be-lro.md:28]: >get [](http://example.com/codeflow?start=16&length=3)

can we maintain consistency with the verb and refer it as GET everywhere in the doc

File: docs/get-must-not-be-lro.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+ARM OpenAPI(swagger) specs
+
+## Related ARM Guideline Code
+
+- RPC-Get-V1-14
+
+## Output Message
+
+The Get call cannot be Long Running Operation and it must not have `x-ms-long-running-operation-options` property block.
```
Comment (AkhilaIlla) [inline docs/get-must-not-be-lro.md:17]: The GET operation cannot be Long Running and MUST not have x-ms-long-running-operation property defined?

File: docs/get-must-not-be-lro.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+## Description
+
+Only asynchronous(i.e. Long Running Operation) can have `x-ms-long-running-operation-options` property.
+The Get calls are synchronous and it MUST NOT have
+    - `x-ms-long-running-operation-options` property block
+    - `x-ms-long-running-operation-options` set to `true`
```
Comment (AkhilaIlla) [inline docs/get-must-not-be-lro.md:24]: >`x-ms-long-running-operation-options [](http://example.com/codeflow?start=6&length=36)

should be x-ms-long-running-operation ?

File: docs/get-must-not-be-lro.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+## Output Message
+
+The Get call cannot be Long Running Operation and it must not have `x-ms-long-running-operation-options` property block.
+
+## Description
+
+Only asynchronous(i.e. Long Running Operation) can have `x-ms-long-running-operation-options` property.
+The Get calls are synchronous and it MUST NOT have
+    - `x-ms-long-running-operation-options` property block
```
Comment (AkhilaIlla) [inline docs/get-must-not-be-lro.md:23]: >block [](http://example.com/codeflow?start=53&length=5)

I dont think its a block, just a property?

Comment (tejaswiMinnu) [reply docs/get-must-not-be-lro.md:23]: x-ms-long-running-operation is property, but x-ms-long-running-operation-options is block

File: docs/get-must-not-be-lro.md (hunk 4 above)
Comment (tejaswiMinnu) [reply docs/get-must-not-be-lro.md]: done

File: docs/get-must-not-be-lro.md (hunk 5 above)
Comment (tejaswiMinnu) [reply docs/get-must-not-be-lro.md]: x-ms-long-running-operation this have to be  x-ms-long-running-operation-options propertyblock

File: packages/rulesets/src/spectral/az-arm.ts (hunk 3 above)
Comment (tejaswiMinnu) [reply packages/rulesets/src/spectral/az-arm.ts]: I found this big while naming the files with hyphen "get-operations-must-not-be-long-running.*", but still considered and changed.

File: docs/get-operation-must-not-be-long-running.md [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+
+ARM OpenAPI(swagger) specs
+
+## Related ARM Guideline Code
+
+- RPC-Get-V1-14
+
+## Output Message
+
+The GET operation cannot be Long Running and it MUST NOT have `x-ms-long-running-operation-options` property block defined.
```
Comment (rkmanda) [inline docs/get-operation-must-not-be-long-running.md:17]: >Long Running [](http://example.com/codeflow?start=28&length=12)

long running - lower case

Comment (rkmanda) [inline docs/get-operation-must-not-be-long-running.md:17]: >MUST NOT [](http://example.com/codeflow?start=48&length=8)

must not - lower case

File: docs/get-operation-must-not-be-long-running.md [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+
```
Comment (rkmanda) [inline docs/get-operation-must-not-be-long-running.md:18]: The GET operation cannot be long running. It must not have the 'x-ms-long-running-operation' and `x-ms-long-running-operation-options` properties defined. It must not have the status codes other than 200 and default.

File: docs/get-operation-must-not-be-long-running.md [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+Only asynchronous(i.e. Long Running Operation) can have `x-ms-long-running-operation-options` property.
+The GET calls are synchronous and it MUST NOT have
+    - `x-ms-long-running-operation-options` property block
+    - `x-ms-long-running-operation` set to `true`
+
+## How to fix the violation
+
+Ensure that the GET operation `x-ms-long-running-operation` property set to `false` and MUST NOT contain `x-ms-long-running-operation-options` property block defined.
+
```
[the remark this answers is not in the document]
Comment (rkmanda) [reply docs/get-operation-must-not-be-long-running.md:29]: for the status code part, pl check if there is already coverage with another rule and ignore my comment about status codes if there is coverage

[the remark this answers is not in the document]
Comment (rkmanda) [reply docs/get-operation-must-not-be-long-running.md:29]: never mind. it is covered by RPC-Get-V1-01

**Review by rkmanda — CHANGES_REQUESTED** [review]
:clock1:

File: docs/get-operation-must-not-be-long-running.md (hunk 9 above)
Comment (tejaswiMinnu) [reply docs/get-operation-must-not-be-long-running.md:18]: I am ignoring status codes part as its already covered by RPC-Get-V1-01

File: docs/get-operation-must-not-be-long-running.md [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+ "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MyNameSpace/MyResourceType/{Name}": {
+    "get": {
+      ...
+      ...
+      "responses": {
+        "200": {
+            ...
+            }
+          }
+      // Remove or set to false
```
[the remark this answers is not in the document]
Comment (tejaswiMinnu) [reply docs/get-operation-must-not-be-long-running.md:43]: Ignoring this part

File: common/changes/@microsoft.azure/openapi-validator-rulesets/tejaswis-getMustNotBeLRO_2023-09-19-23-56.json [hunk 12]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@microsoft.azure/openapi-validator-rulesets",
+      "comment": "Added GetOperationMustNotBeLongRunning, this rule makes sure GET calls must not have LRO properties as GET calls are synchronous",
```
Comment (rkmanda) [inline common/changes/@microsoft.azure/openapi-validator-rulesets/tejaswis-getMustNotBeLRO_2023-09-19-23-56.json:5]: >Added [](http://example.com/codeflow?start=18&length=5)

Simple present tense - Add

**Review by rkmanda — APPROVED** [review]
:shipit:

**Pull request merged**